src/wasm/wasm-binary.cpp

```diff
@@ -486,14 +489,49 @@ void WasmBinaryWriter::writeLateUserSections() {
 }
 
 void WasmBinaryWriter::writeUserSection(const UserSection& section) {
-  auto start = startSection(0);
+  auto start = startSection(BinaryConsts::User);
   writeInlineString(section.name.c_str());
   for (size_t i = 0; i < section.data.size(); i++) {
     o << uint8_t(section.data[i]);
   }
   finishSection(start);
 }
 
+void WasmBinaryWriter::writeFeaturesSection() {
+  if (!wasm->emitFeaturesSection || wasm->features <= FeatureSet{}) {
```

Comment:
does the right side of the if mean "if MVP, don't emit the features section"?

Reply:
Yes, I forgot about the `isMVP()` method which I should definitely use here.